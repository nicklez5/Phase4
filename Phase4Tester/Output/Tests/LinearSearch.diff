--- SelfTestCases/LinearSearch.out	2011-04-28 14:55:21.000000000 -0700
+++ ./Output/Tests/LinearSearch.out	2020-03-14 21:50:36.722514280 -0700
@@ -9,8 +9,8 @@
 18
 9999
 0
-1
-1
+0
+0
 0
 55
 
