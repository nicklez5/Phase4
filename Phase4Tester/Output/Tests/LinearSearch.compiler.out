.data

vmt_LS:
  LS.Start
  LS.Print
  LS.Search
  LS.Init

.text

  jal Main
  li $v0 10
  syscall

Main:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 8
  sw $ra -4($fp)
  li $a0 12
  jal _heapAlloc
  move $t0 $v0
  la $t9 vmt_LS
  sw $t9 0($t0)
  bnez $t0 null1
  la $a0 _str0
  j _error
null1:
  lw $t1 0($t0)
  lw $t1 0($t1)
  move $a0 $t0
  li $a1 10
  jalr $t1
  move $t1 $v0
  move $a0 $t1
  jal _print
  lw $ra -4($fp)
  lw $fp -8($fp)
  addu $sp $sp 8
  jr $ra

LS.Start:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 12
  sw $ra -4($fp)
  sw $s0 0($sp)
  move $s0 $a0
  move $t0 $a1
  lw $t1 0($s0)
  lw $t1 0($t1)
  move $a0 $s0
  move $a1 $t0
  jalr $t1
  lw $t1 0($s0)
  lw $t1 0($t1)
  move $a0 $s0
  jalr $t1
  li $a0 9999
  jal _print
  lw $t1 0($s0)
  lw $t1 0($t1)
  move $a0 $s0
  li $a1 8
  jalr $t1
  move $t1 $v0
  move $a0 $t1
  jal _print
  lw $t1 0($s0)
  lw $t1 0($t1)
  move $a0 $s0
  li $a1 12
  jalr $t1
  move $t1 $v0
  move $a0 $t1
  jal _print
  lw $t1 0($s0)
  lw $t1 0($t1)
  move $a0 $s0
  li $a1 17
  jalr $t1
  move $t1 $v0
  move $a0 $t1
  jal _print
  lw $t1 0($s0)
  lw $t1 0($t1)
  move $a0 $s0
  li $a1 50
  jalr $t1
  move $t1 $v0
  move $a0 $t1
  jal _print
  li $v0 55
  lw $s0 0($sp)
  lw $ra -4($fp)
  lw $fp -8($fp)
  addu $sp $sp 12
  jr $ra

LS.Print:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 8
  sw $ra -4($fp)
  move $t0 $a0
  li $t1 1
while1_top:
  lw $t2 0($t0)
  slti $t2 $t1 $t2
  beqz $t2 while1_end
  lw $t2 0($t0)
  sw $t9 0($t0)
  bnez $t0 null2
  la $a0 _str0
  j _error
null2:
  lw $t3 0($t2)
  move $a0 $t1
  move $a1 $t3
  sw $t9 0($t0)
  bnez $t0 bounds1
  j _error
bounds1:
  mul $t3 $t1 4
  addu $t3 $t3 $t2
  lw $t3 0($t3)
  move $a0 $t3
  jal _print
  addu $t1 $t1 1
  j while1_top
while1_end:
  li $v0 0
  lw $ra -4($fp)
  lw $fp -8($fp)
  addu $sp $sp 8
  jr $ra

LS.Search:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 8
  sw $ra -4($fp)
  move $t0 $a0
  move $t1 $a1
  li $t2 1
  li $t3 0
while2_top:
  lw $t4 0($t0)
  slti $t4 $t2 $t4
  beqz $t4 while2_end
  lw $t4 0($t0)
  sw $t9 0($t0)
  bnez $t0 null3
  la $a0 _str0
  j _error
null3:
  lw $t5 0($t4)
  move $a0 $t2
  move $a1 $t5
  sw $t9 0($t0)
  bnez $t0 bounds2
  j _error
bounds2:
  mul $t5 $t2 4
  addu $t5 $t5 $t4
  lw $t5 0($t5)
  addu $t4 $t1 1
  slti $t6 $t5 $t1
  beqz $t6 if1_else
  j if1_end
if1_else:
  slti $t4 $t5 $t4
  subu $t4 1 $t4
  beqz $t4 if2_else
  j if2_end
if2_else:
  li $t3 1
  lw $t2 0($t0)
if1_end:
  addu $t2 $t2 1
  j while2_top
while2_end:
  move $v0 $t3
  lw $ra -4($fp)
  lw $fp -8($fp)
  addu $sp $sp 8
  jr $ra

LS.Init:
  sw $fp -8($sp)
  move $fp $sp
  subu $sp $sp 12
  sw $ra -4($fp)
  sw $s0 0($sp)
  move $s0 $a0
  move $t0 $a1
  move $s0 $v0
